service/controller/resource/app/delete.go

```diff
@@ -98,6 +92,86 @@ func (r Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
+func (r Resource) cancel(ctx context.Context) error {
+	finalizerskeptcontext.SetKept(ctx)
+
+	r.logger.Debugf(ctx, "keeping finalizers")
+	r.logger.Debugf(ctx, "canceling resource")
+
+	return nil
+}
+
+// deleteClusterApps tries to delete given apps, skipping apps with
+// Flux managed-by label.
+func (r Resource) deleteClusterApps(ctx context.Context, apps []*v1alpha1.App) error {
+	for _, app := range apps {
+		managedBy := appkey.ManagedByLabel(*app)
+
+		// No need to delete app whose deletion has already been requested,
+		// or when managed by Flux as the app may be recreated in such case.
+		// There is one valid case when deleting Flux-managed app makes sense,
+		// namely when `prune: false` is used and app is gone in the repository,
+		// but there is no way to recognize this case here.
+		if managedBy == fluxManagedByLabel {
```

Comment:
App CRs comming from bundles most probably won't carry this, though the bundle itself still may, as it can be created by Flux and may, after removal, be recreated by Flux, hence skipping it here. Same apply to non-bundled App CRs. I don't understand the word "suddently", I see nothing sudden in this personally, there is no a single and "right" channel to deliver the apps, and Flux is one of the ways we recommend the most. I may decide to skip deletion of App managed by it,  but it feels a bit weird when Flux takes care of them, and I'm not able to precisely predict the outcome then, see https://github.com/giantswarm/giantswarm/issues/20751#issuecomment-1145900518.